bin/geonames-sql.js

```diff
@@ -13,98 +12,142 @@ import slug from 'slug';
 let exec_env = process.env.DB_ENV || 'development';
 let root = fs.realpathSync(`${__dirname}/..`);
 
-let Knex = knex(require(`${root}/knexfile.js`)[exec_env]);
+let knex = Knex(require(`${root}/knexfile.js`)[exec_env]);
+
+/**
+ * @param {string} table
+ * @param {object} condition
+ * @param {object} attributes
+ * @returns {string} - sql query string
+ */
+async function createOrUpdate(table, condition, attributes) {
+  let result = await knex(table).where(condition);
+  let query;
+
+  if (result.length > 0) {
+    query = knex(table)
+      .where(condition)
+      .update(attributes)
+      .toString();
+  } else {
+    query = knex(table)
+      .insert(attributes)
+      .toString();
+  }
+
+  // Knex escapes double quotes that don't need to be escaped (e.g. when using JSON.stringify).
+  return query.replace(/\\\\"/g, '\\"') + ';';
```

Comment:
1. is there a bug for this?
2. did you check knex 0.10? there was something in changelog about escaping

I just feel this is very fragile.